src/components/process/manage/ManageStages.js

```diff
@@ -0,0 +1,78 @@
+import {useEffect, useState} from "react";
+import {processAPI} from "../../../utils/apis/ProcessAPI";
+import {Box, Card, CardContent, Typography} from "@material-ui/core";
+import {TwoColumnDnD} from "estella-two-column-dnd";
+import {colors} from "../../../utils/colors";
+import {withSwal} from "../../formsCommons/WithSwal";
+import Swal from "sweetalert2";
+
+export const ManageStages = ({processId}) => {
+
+    const [stages, setStages] = useState([])
+    const [possibleStages, setPossibleStages] = useState([])
+
+    useEffect(() => {
+        let swal = new Swal({
+            title: "Getting data"
+        })
+        Swal.showLoading()
+        processAPI.getProcessById(processId)
+            .then(data => {
+                setStages(data.stages);
+                swal.close()
+            })
+        processAPI.getAllPossibleStages()
+            .then(data => setPossibleStages(data))
+    }, [processId])
+
+    const getPossibleStages = () => possibleStages.map(stage => {
+        return {'type': stage}
+    })
+
+    const getStages = () => stages
+
+    const handleSubmit = (items) => {
+        withSwal({
+            loadingTitle: "Updating stages",
+            promise: () => processAPI.updateProcessStages(processId, items.map(item => item.type)),
+            successSwalTitle: "Stages Updated"
+        })
+    }
+
+    const fireSwal = (title, text, icon) => {
+        Swal.fire({
+            title: title,
+            text: text,
+            icon: icon
+        })
+    }
+
+    return (
+        <TwoColumnDnD
+            firstListItems={getStages()}
+            secondListItems={getPossibleStages()}
+            forbiddenIndexes={[0, getStages()?.length]}
+            warningFunction={() => fireSwal(
+                "You can't do this!",
+                "We're sorry, but you cannot set this stage here! See help for more information.",
+                "warning"
+            )}
+            itemRender={(item) => (
+                <Box m={1}>
+                    <Card variant="outlined" style={{textAlign: "center"}}>
+                        <CardContent>
+                            <Typography>{item.type}</Typography>
+                        </CardContent>
+                    </Card>
+                </Box>
+            )}
+            onSubmit={(first, _) => handleSubmit(first)}
```

Comment:
Pytanko, czy potrzebne jest to podkreślenie, wydaje mi się że w js jak podasz do funkcji za dużo argumentów to po prostu wywołuje z tyloma ile się da. Ale to nwm, bo się nie znam

Reply:
Możliwe, że tak, ale to wziąłem z kotlina + może być to info, że jest tam dostęp do drugiego argumentu na przyszłość.